### PR TITLE
Add "inductor_pre_grad_graph" logging

### PR DESCRIPTION
**Pull request by yushangdi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #142717

Add new structured logging "inductor_pre_grad_graph"

This is for inductor provenance tracking front-end to load this graph from tlparse.

Differential Revision: [D67054578](https://our.internmc.facebook.com/intern/diff/D67054578/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/compile_fx.py [hunk 1]
```diff
@@ -1633,6 +1634,24 @@ def compile_fx(
...
+                "inductor_pre_grad_graph",
+                payload_fn=lambda: model_.print_readable(
+                    print_output=False, include_stride=True, include_device=True
+                )
+                + f"\n\n # graph id: {id(model_.graph)}",
+            )
+            pre_grad_graphs_log.debug(
+                "%s",
+                lazy_format_graph_code(
+                    "BEFORE GRAD",
```
Comment (desertfire) [inline torch/_inductor/compile_fx.py:1647]: Let's add "PRE" here.

File: torch/_logging/_registrations.py [hunk 2]
```diff
@@ -97,6 +97,10 @@
     "Prints the FX forward and backward graph generated by AOTDispatch, useful for debugging effects processing.",
     visible=True,
 )
+register_artifact(
+    "pre_grad_graphs",
+    "Prints the FX graph before inductor grad passes. Useful to understand what's being given to Inductor before grad passes",
```
Comment (desertfire) [inline torch/_logging/_registrations.py:102]: Also better to spell "pre" out here.

**Review by desertfire — APPROVED** [review]

**Comment by desertfire** [comment]
https://github.com/pytorch/pytorch/pull/137803#pullrequestreview-2363724748  , following this example, did you update tlparse to understand the new traced info?

**Comment by yushangdi** [comment]
> [#137803 (review)](https://github.com/pytorch/pytorch/pull/137803#pullrequestreview-2363724748) , following this example, did you update tlparse to understand the new traced info?

@desertfire  Yeah, I've updated the tlparse

**Comment by yushangdi** [comment]
Close because of CLA issue, re-submitted the same PR in https://github.com/pytorch/pytorch/pull/143126

**Pull request closed without merging**